### DISCUSSION TITLE
Importing data to the energy dashboard

### DISCUSSION THREAD

**Discussion by Thepyropc** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#118] [Q&A]
so im not sure where to put the code for kwh
#Total kWh
  - platform: total_daily_energy
    name: ${disp_name} Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh
time:
  - platform: sntp
    id: sntp_time 

like does it go into the programing in esp home or does it go in my configuration file on home assistant im just confused cause wherever i put it i get errors

**Comment by CircuitSetup** [comment]
It should go in the esphome config for the energy meter esp32. [See the example here.](https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/blob/e59a9eae6e5ebcf630f7cea400272ee6fea57777/Software/ESPHome/6chan_energy_meter_main_board.yaml#L166)

**Reply by Thepyropc** [reply]
![Screenshot_20221031_080324_Home Assistant](https://user-images.githubusercontent.com/90295842/199014681-1c61d597-9b29-4dc3-9de3-72e24f2917bb.jpg)
![Screenshot_20221031_080304_Home Assistant](https://user-images.githubusercontent.com/90295842/199014686-2f17770b-10f9-4f0c-9d8e-e9b0257f7550.jpg)

I cant get passed a validation

**Reply by CircuitSetup** [reply]
> I cant get passed a validation

It looks like your spacing is incorrect in the config file. Make sure to copy directly from Github or the downloaded example file.

**Comment by Thepyropc** [comment]
![Screenshot_20221031_085321_Home Assistant](https://user-images.githubusercontent.com/90295842/199024368-1adcb745-b9a2-434e-a901-c00941e22313.jpg)


Formated the same way
![Screenshot_20221031_085429_Home Assistant](https://user-images.githubusercontent.com/90295842/199024546-2ec6d8b7-09e5-4d7a-85cb-f4b75fc2063e.jpg)

**Reply by CircuitSetup** [reply]
add this to the end: 

```
time:
  - platform: sntp
    id: sntp_time   
```

**Comment by Thepyropc** [comment]
![Screenshot_20221031_092434_Home Assistant](https://user-images.githubusercontent.com/90295842/199031142-9c119efd-8710-4707-8cfa-48f52e6d6d6f.jpg)
![Screenshot_20221031_092418_Home Assistant](https://user-images.githubusercontent.com/90295842/199031147-5aa730e8-bd5d-4577-ac95-cd28e56d856e.jpg)
STill giving me some errors

Do u want me to send the whole code? im running out of ideas

**Reply by CircuitSetup** [reply]
Right, you need the previous part in the example too...
```
#Total Watts
  - platform: template
    name: Energy Meter Total Watts
    id: totalWatts
    lambda: return id(ct1Watts).state + id(ct2Watts).state + id(ct3Watts).state + id(ct4Watts).state + id(ct5Watts).state + id(ct6Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: energy
    update_interval: 5s
```

Make sure to change ct1watts, ct2watts, etc to what you have for the phase > power > id: if the IDs were changed.

**Reply by Thepyropc** [reply]
# Example CircuitSetup 6-channel entry
spi:
  clk_pin: 18
  miso_pin: 19
  mosi_pin: 23
sensor:
  - platform: atm90e32
    cs_pin: 5
    phase_a:
      voltage:
        name: "Voltage A"
      current:
        name: "ComEd A Amps"
      power:
        name: "ComeEd A Watts"
      gain_voltage: 7380
      gain_ct: 27518
    phase_b:
      current:
        name: "ComEd B Amps"
      power:
        name: "ComEd B Watts"
      gain_voltage: 7380
      gain_ct: 27518
    phase_c:
      current:
        name: "To Garage A Amps"
      power:
        name: "To Garage A Watts"
      gain_voltage: 7380
      gain_ct: 27518
    frequency:
      name: "EMON Line Frequency"
    line_frequency: 50Hz
    current_phases: 3
    gain_pga: 1X
    update_interval: 5s
  - platform: atm90e32
    cs_pin: 4
    phase_a:
      current:
        name: "To Garage B Amps"
      power:
        name: "To Garage B Watts"
      gain_voltage: 7380
      gain_ct: 29000
    phase_b:
      current:
        name: "From Garage A Amps"
      power:
        name: "Fron Garage A Watts"
      gain_voltage: 7380
      gain_ct: 41660
    phase_c:
      current:
        name: "From Garage B Amps"
      power:
        name: "Fron Garage B Watts"
      gain_voltage: 7380
      gain_ct: 41660
    line_frequency: 50Hz
    current_phases: 3
    gain_pga: 1X
    update_interval: 5s

#Total Watts
  - platform: template
    name: Energy Meter Total Watts
    id: totalWatts
    lambda: return id(ComeEd A Watts).state + id(ComEd B Watts)
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: energy
    update_interval: 5s
#kWh
  - platform: total_daily_energy
    name: ${disp_name} Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh
    device_class: energy
    state_class: total_increasing

time:
  - platform: sntp
    id: sntp_time  


when i validate it it works but when i push it to install it does not it gives a bunch of errors (sorry lol)

**Reply by Thepyropc** [reply]
ok i rewrote it all but now i am getting one final error and im not sure how to get passed it cause it shouldnt be giving it to me i dont think

here's the code
`captive_portal:

substitutions:
  disp_name: 6C
  update_time: 10s
  current_cal: '27961'

spi:
  clk_pin: 18
  miso_pin: 19
  mosi_pin: 23
sensor:
  - platform: atm90e32
    cs_pin: 5
    phase_a:
      voltage:
        name: ${disp_name} Volts A
        id: ic1Volts
        accuracy_decimals: 1
      current:
        name: ComEd A Amps
        id: ct1Amps
      gain_voltage: 7380
      gain_ct: 27518
    phase_b:
      current:
        name: ComEd B Amps
        id: ct2Amps
      gain_ct: 27518
    phase_c:
      current:
        name: To Garage A Amps
        id: ct3Amps
      gain_ct: 27518
    frequency:
      name: ${disp_name} Freq A
    line_frequency: 60Hz
    current_phases: 3
    gain_pga: 1X
    update_interval: ${update_time}
  - platform: atm90e32
    cs_pin: 4
    phase_a:
      voltage:
        name: ${disp_name} Volts B
        id: ic2Volts
        accuracy_decimals: 1
      current:
        name: To Garage B Amps
        id: ct4Amps
      gain_voltage: 7380
      gain_ct: 29000
    phase_b:
      current:
        name: From Garage A Amps
        id: ct5Amps
      gain_ct: 41660
    phase_c:
      current:
        name:From Garage B Amps
        **id**: ct5Amps
      gain_ct: 41660
    frequency:
      name: ${disp_name} Freq B
    line_frequency: 60Hz
    current_phases: 3
    gain_pga: 1X
    update_interval: 5s

#Watts per channel
  - platform: template
    name: ComEd A Watts
    id: ct1Watts
    lambda: return id(ct1Amps).state * id(ic1Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
  - platform: template
    name: ComEd B Watts
    id: ct2Watts
    lambda: return id(ct2Amps).state * id(ic1Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
  - platform: template
    name: To Garage A Watts
    id: ct3Watts
    lambda: return id(ct3Amps).state * id(ic1Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
  - platform: template
    name: To Garage B Watts
    id: ct4Watts
    lambda: return id(ct4Amps).state * id(ic2Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
  - platform: template
    name: From Garage A Watts
    id: ct5Watts
    lambda: return id(ct5Amps).state * id(ic2Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
  - platform: template
    name: From Garage B Watts
    id: ct6Watts
    lambda: return id(ct6Amps).state * id(ic2Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s
#Total Amps
  - platform: template
    name: ${disp_name} Total Amps
    id: totalAmps
    lambda: return id(ct1Amps).state + id(ct2Amps).state + id(ct3Amps).state + id(ct4Amps).state + id(ct5Amps).state + id(ct6Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}

#Total Failover Amps
  - platform: template
    name: ${disp_name} Total Amps
    id: totalFailoverAmps
    lambda: return id(ct5Amps).state + id(ct6Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: 5s

#Total Watts
  - platform: template
    name: ComEd Total Watts
    id: totalWatts
    lambda: return id(totalAmps).state * id(ic1Volts).state;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s

#Total Failover Watts
  - platform: template
    name: Failover Total Watts
    id: totalFailoverWatts
    lambda: return id(totalFailoverAmps).state * id(ic1Volts).state;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: 5s

#kWh
  - platform: total_daily_energy
    name: ComEd Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh`



The line 
        id: ct5Amps
is giving me an error mapping values are not allowed here (i tried to bold it in the code) and when i validate it or try to install it fails and wont let me not sure what to do

**Comment by CircuitSetup** [comment]
You have ct5Amps declared twice. For phase_c (From Garage B Amps) it should be ct6Amps.

Also, you do not need templates to calculate Watts. The meter will do this more accurately with 'power'. Again, see the example. If you get a negative value with power, turn the CT around on the wire.

**Comment by Thepyropc** [comment]
![image](https://user-images.githubusercontent.com/90295842/200138015-d9ca35ad-ad2d-4844-9f7a-bb29e606fd25.png)

this is what it is doing it is just giving it to me on the ID part

**Reply by Thepyropc** [reply]
![image](https://user-images.githubusercontent.com/90295842/200138061-f04cca66-02e6-4509-9f6a-81a803731fdf.png)
